src/main/java/org/jabref/gui/entryeditor/FieldsEditorTab.java

```diff
@@ -0,0 +1,326 @@
+import javafx.scene.control.Label;
+import javafx.scene.layout.ColumnConstraints;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.RowConstraints;
+
+import org.jabref.Globals;
+import org.jabref.gui.BasePanel;
+import org.jabref.gui.FXDialogService;
+import org.jabref.gui.GUIGlobals;
+import org.jabref.gui.JabRefFrame;
+import org.jabref.gui.fieldeditors.FieldEditorFX;
+import org.jabref.gui.fieldeditors.FieldEditors;
+import org.jabref.gui.fieldeditors.FieldNameLabel;
+import org.jabref.gui.keyboard.KeyBinding;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.entry.FieldProperty;
+import org.jabref.model.entry.InternalBibtexFields;
+
+/**
+ * A single tab displayed in the EntryEditor holding several FieldEditors.
+ */
+class FieldsEditorTab extends EntryEditorTab {
+
+    private final Region panel;
+    private final List<String> fields;
+    private final EntryEditor parent;
+    private final Map<String, FieldEditorFX> editors = new LinkedHashMap<>();
+    private final JabRefFrame frame;
+    private final BasePanel basePanel;
+    private FieldEditorFX activeField;
+    private BibEntry entry;
+
+    public FieldsEditorTab(JabRefFrame frame, BasePanel basePanel, List<String> fields, EntryEditor parent,
+                           boolean addKeyField, boolean compressed, BibEntry entry) {
+        this.entry = Objects.requireNonNull(entry);
+        this.fields = new ArrayList<>(Objects.requireNonNull(fields));
+
+        // Add the edit field for Bibtex-key.
+        if (addKeyField) {
+            this.fields.add(BibEntry.KEY_FIELD);
+        }
+
+        this.parent = parent;
+        this.frame = frame;
+        this.basePanel = basePanel;
+
+        panel = setupPanel(frame, basePanel, compressed);
+
+        // The following line makes sure focus cycles inside tab instead of being lost to other parts of the frame:
+        //panel.setFocusCycleRoot(true);
+    }
+
+    private static void addColumn(GridPane gridPane, int columnIndex, List<Label> nodes) {
+        gridPane.addColumn(columnIndex, nodes.toArray(new Node[nodes.size()]));
+    }
+
+    private static void addColumn(GridPane gridPane, int columnIndex, Stream<Parent> nodes) {
+        gridPane.addColumn(columnIndex, nodes.toArray(Node[]::new));
+    }
+
+    private String convertToHex(java.awt.Color color) {
+        return String.format("#%02x%02x%02x", color.getRed(), color.getGreen(), color.getBlue());
+    }
+
+    private Region setupPanel(JabRefFrame frame, BasePanel bPanel, boolean compressed) {
+
+        //setupKeyBindings(panel.getInputMap(JComponent.WHEN_FOCUSED), panel.getActionMap());
+
+        editors.clear();
+        List<Label> labels = new ArrayList<>();
+
+        for (String fieldName : fields) {
+
+            // TODO: Reenable/migrate this
```

Comment:
I actually didn't change anything here in this PR (github just marks the whole file as changed since it was renamed).